src/main/java/org/folio/rest/impl/SamlAPI.java

```diff
@@ -293,20 +309,67 @@ private Future<JsonObject> getUsersResponse(WebClient webClient, SamlConfigurati
     final String usersCql = getCqlUserQuery(userPropertyName, samlAttributeValue);
     final String userQuery = UriBuilder.fromPath("/users").queryParam("query", usersCql).build().toString();
 
-    return webClient.getAbs(parsedHeaders.getUrl() + userQuery)
-      .putHeader(XOkapiHeaders.TOKEN, parsedHeaders.getToken())
-      .putHeader(XOkapiHeaders.URL, parsedHeaders.getUrl())
-      .putHeader(XOkapiHeaders.TENANT, parsedHeaders.getTenant())
+    return handleGetUser(userPropertyName, samlAttributeValue, webClient, parsedHeaders)
```

Comment:
Lets extract these 2 methods, getUser and getUser tenant into separate service, in this case we will simplify SamlAPI and our implementation would be similar with mod-login